projects/libyaml/libyaml_reformatter_fuzzer.c

```diff
@@ -0,0 +1,90 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "yaml.h"
+#include "yaml_write_handler.h"
+#include <assert.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#ifdef NDEBUG
+#undef NDEBUG
+#endif
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+  if (size < 2)
+    return 0;
+
+  bool done = false;
+  bool is_canonical = data[0] & 1;
+  bool is_unicode = data[1] & 1;
+  data += 2;
+  size -= 2;
+
+  yaml_parser_t parser;
+  yaml_emitter_t emitter;
+  yaml_event_t event;
+  FILE *f;
+
+  /* Initialize the parser and emitter objects. */
+
+  if (!yaml_parser_initialize(&parser))
+    goto error;
+
+  if (!yaml_emitter_initialize(&emitter))
+    goto error;
```

Comment:
make this i goto cleanup_parser or similar (and have a label at the end before yaml_parser_delete